### ISSUE TITLE
Invalid code is silently generated for varargs function

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#5126]
|  |  |
| --- | --- |
| Bugzilla Link | [4754](https://llvm.org/bz4754) |
| Version | unspecified |
| OS | Linux |
| Attachments | [C and LL files](https://user-images.githubusercontent.com/60944935/143739723-601a015c-a0e1-4057-9f3a-ce2537245748.gz) |
| Reporter | LLVM Bugzilla Contributor |
| CC | @asl |

## Extended Description 

Both LLVM-GCC and CLANG are affected. "__attribute__((__stdcall__))" does not cause errors. FSF GCC silently ignores "__attribute__((__fastcall__))".

LLVM r79531.

**Comment by gwelymernans** [comment]
Clang does warn about this (and the program runs correctly).

```
$ clang -c a.c -O2
a.c:2:25: warning: '__fastcall__' calling convention is not supported for this target [-Wignored-attributes]
    int (__attribute__((__fastcall__)) *f)();
                        ^
1 warning generated.

$ clang -c b.c -O2
b.c:2:25: warning: '__fastcall__' calling convention is not supported for this target [-Wignored-attributes]
    int (__attribute__((__fastcall__)) *f)();
                        ^
b.c:5:20: warning: '__fastcall__' calling convention is not supported for this target [-Wignored-attributes]
int __attribute__((__fastcall__)) bar(int a, int b, int c, int d)
                   ^
2 warnings generated.

$ clang a.o b.o -o a
$ ./a
$ echo $?
46
```
